flow-typed/npm/redux-loop_v4.x.x.js

```diff
@@ -0,0 +1,176 @@
+// @flow
+
+import type { ActionCreator, StoreEnhancer, Store } from "redux";
+
+declare module "redux-loop" {
+  declare export type StoreCreator = <S, A>(
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ) => Store<S>;
+
+  declare export type StoreCreatorExplicit<S, A> = (
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ) => Store<S>;
+
+  declare export type Loop<S, A> = [S, CmdType<A>];
+
+  declare export type LoopReducer<S, A> = (
+    state: S | void,
+    action: A,
+    ...args: any[]
+  ) => Loop<S, A> | S;
+
+  declare export interface LiftedLoopReducer<S, A> {
+    (state: S | void, action: A, ...args: any[]): Loop<S, A>;
+  }
+
+  declare export type CmdSimulation<R> = {
+    result: R,
+    success: boolean
+  };
+
+  declare export interface MultiCmdSimulation {
+    [index: number]: CmdSimulation<any> | MultiCmdSimulation;
+  }
+
+  declare export interface NoneCmd {
+    type: "NONE";
+    simulate(): null;
+  }
+
+  declare export interface ListCmd<A> {
+    type: "LIST";
+    cmds: CmdType<A>[];
+    sequence?: boolean;
+    batch?: boolean;
+    simulate(simulations: MultiCmdSimulation): A[];
+  }
+
+  declare export interface ActionCmd<A> {
+    type: "ACTION";
+    actionToDispatch: A;
+    simulate(): A;
+  }
+
+  declare export interface MapCmd<A, B = any, C = any> {
+    type: "MAP";
+    tagger: ((...args:Array<C|B>) => A) | ((subAction: B) => A);
+    nestedCmd: CmdType<A>;
+    args: Array<C>;
+    simulate(simulations?: CmdSimulation<any> | MultiCmdSimulation): A[] | A | null;
+  }
+
+  declare export interface RunCmd<A, B, C> {
+    type: "RUN";
+    func: (...args: Array<C>) => B;
+    args?: Array<C>;
+    failActionCreator?: ActionCreator<A, any>;
+    successActionCreator?: ActionCreator<A, B>;
+    forceSync?: boolean;
+    simulate(simulation: CmdSimulation<B>): A;
+  }
+
+  declare export type SequenceCmd<A> = ListCmd<A>;
```

Comment:
i'd rather not export these since they're deprecated anyway. Since this is new, no need to support deprecated stuff

Reply:
ok. I mean this is direct translation of TS signatures

Reply:
yea typescript was done before those were deprecated.